File: SliferSPARQL.js
var request = require("superagent"),
	stdin = require('stdin'),
	utf8 = require('utf8');
	
var outputSSparql = [];


stdin(function(chunk){
  analyzeData(chunk);
});

function analyzeData(chunk)
{
	var URLIs = JSON.parse(chunk);
	
	URLIs.forEach(function(tab)
	{
		var query = 'SELECT * WHERE { ?s ?p ?o. FILTER(?s in (';
		
		tab.resources.forEach(function(uri){
			query = query + '<' + uri + '>,';
		});
		
		query = query.slice(0,-1);
		query = query + ')) }';
		
		request
			.post('http://live.dbpedia.org/sparql')
			.send('default-graph-uri='+encodeURI('http://dbpedia.org'))
			.send('query='+encodeURI(query))
			.send('format=json')
			.send('timeout=30000')
			.buffer(true)
			.set('Accept', '*')
			.end(function(res){
				console.log("1");
				//parseData(res.text);
			});

	});
}

function parseData(data)
{
	var res = JSON.parse(data);
	
	var output = [];
	
	res.results.bindings.forEach(function(elt, i)
	{
		if (elt.o.type != 'uri') {
			return;
		}
		
		output.push({
			s: elt.s.value,
			p: elt.p.value,
			o: elt.o.value
		});
		
	});
	
	console.log(output);
	
}

File: SliferSPARQL.py
#!/usr/bin/python
import re
import json
import sys
import ast
import urllib2
import urllib
import unicodedata
import re 

final = []
def clean(data) : 
	data =  data.replace("\U","")
	return data

def parseData(data):
	res = json.loads(data)
	output = []

	for elt in res["results"]["bindings"]:
		if elt["o"]["type"] != 'uri' : 
			continue
		myelem = {
			"s" : elt["s"]["value"],
			"p" : elt["p"]["value"],
			"o" : elt["o"]["value"]
			}

		output.append(myelem)

	return output

if __name__ == '__main__':
	final = []


	jsonArg = sys.stdin.read()  
	
	jsonArg = ast.literal_eval(jsonArg)


	for URI in jsonArg: 
		url = URI['url'];
		resources = URI['resources']
		#print url
		#print resources

		d = 'SELECT * WHERE { ?s ?p ?o. FILTER(?s in ('
		for resource in resources : 
			d = d + '<' + resource + '>,'
		d = d[:-1]
		d = d + ')) }'

		values = {
			"default-graph-uri" : "http://dbpedia.org",
			"format" : "json",
			"timeout" : "30000", 
			"query" : d
		}
		header = { "Accept" : "*" }

		urlReq = "http://dbpedia.org/sparql"
		data = urllib.urlencode(values)
		req = urllib2.Request(urlReq,data,header)
		response = urllib2.urlopen(req)
		html = response.read()

		html =  clean(html)
		parsed = parseData(html)
		finalParsed = {
			"url" : url,
			"results" : parsed
		}
		final.append( finalParsed )

	json.dump(final, sys.stdout, sort_keys = False, indent = 4)




